### DISCUSSION TITLE
truncate等函数第二个参数优化

### DISCUSSION THREAD

**Discussion by arvin-777** [apache/doris#32086] [A - General / Q&A]
像postgrel里的去精度字段 trunc(change, prec_show)这样写就可以了，第二个参数可以动态的指定某一列，doris好像很多函数的第二个参数都是必须写死，比如truncate函数的第二个参数必须是常量，如果我某条数据的精度必须关联币种表设置的精度去截取就不能用这个函数，我要达到截取的目的只能    greatest(cast( concat(split_part(a.change, '.', 1),'.',strleft(split_part(a.change, '.', 2),c.prec_show)) as decimal(38,20)),0) AS change

**Comment by zhiqiang-hhhh** [comment]
https://github.com/apache/doris/pull/32746

**Reply by zhiqiang-hhhh** [reply]
这个 PR 加强了 truncate，truncate 之后可以支持使用 column 作为 scale 参数，对于 decimal 来说，结果的精度取决于 scale 的形式，如果是 literal，那么结果 scale 取决于 scale 参数；如果使用 column 作为 scale ，那么结果的精度始终与 input decimal 相同
